syntax = "proto3";
package directmq.v1;
option go_package = "./protocol";

import "directmq/v1/connection.proto";
import "directmq/v1/publish.proto";
import "directmq/v1/subscribe.proto";
import "directmq/v1/unsubscribe.proto";

message DataFrame {
    int32 ttl = 1;
    repeated string traversed = 2;

    oneof message {
        SupportedProtocolVersions supported_protocol_versions = 3;
        InitConnection init_connection = 4;
        ConnectionAccepted connection_accepted = 5;
        Publish publish = 6;
        Subscribe subscribe = 7;
        Unsubscribe unsubscribe = 8;
        GracefullyClose gracefully_close = 9;
        TerminateNetwork terminate_network = 10;
    }
}
